### DISCUSSION TITLE
subtitles / captions from youtube?

### DISCUSSION THREAD

**Discussion by gregconquest** [aclap-dev/video-downloadhelper#885] [Help and Q&A]
### Your question

There are websites that can download captions/subtitles for youtube videos -- in .srt format. Does VDH not do this? It would be very convenient if VDH could download both the video and the subtitles together (as .mp4 and .srt). I'm surprised that no one has already asked this question.

### Addon & CoApp version and details

version: 9.0.2.3
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp","version":"2.0.19","new_version":false}
license: {"unset":true}
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:127.0) Gecko/20100101 Firefox/127.0
{
    "used_sidebar_button": true,
    "used_popup_button": true,
    "download_directory": "/Users/gregconquest/media_editing/me_ecques",
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_sidebar": false,
    "open_count_store": 3,
    "successfull_dl": 1
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

Surprised?  That may be because you have not searched our old forum on Google Groups . . . which just happened to get resurrected today after being banned for the past couple of weeks.  I don't know how the decision was reached to resurrect it but it's there now.  It will probably get banned again for no good reason a few weeks or months from now.  That's why we moved here to GitHub.  They aren't quite so frivolous here.  In any case, VDH has never accessed subtitles.  @paulrouget Paul, our developer, has said in public that support for captions in VDH is very much on his agenda.  But I haven't seen his agenda so I have no idea how soon this much-requested feature will appear in the product.

There is a thread on our old forum with the title:

How to download from YouTube

May as well pimp it a bit during this probably narrow window that the group is back on the air.  The techniques described there can be used to locate YouTube subtitles.  My observation was that not everything on YouTube has captions but when they do exist, you can find them using those techniques.  The trick is that they are in some weird format that I am entirely unfamiliar with.  Probably yet another abuse of monopoly power by Google.  They do things their own way & ignore existing international standards.  I personally prefer VTT & I am familiar with SRT.  So on the very rare occasions I have retrieved YouTube subtitles, I have processed them through a little freebie program called Subtitle Edit.  You can find it via web search, which is how I found it.  Actually, it's not so little.  It has an astonishing array of features, of which I probably have ever used only 3%.  But it can take YouTube captions as input & generate subtitles in VTT, SRT, & I imagine a number of other formats.  Amazing tool, especially considering it's free.  Although you should quickly feel guilt about it being free & send the guy a donation.  He is astonishingly active in his support forum here on GitHub & the rate he releases new beta versions will make your head spin.

So if you are particularly ambitious & persevering, you can get captions from YouTube yourself today.  If you want to wait until VDH can do it, well, I wouldn't recommend holding your breath.

**Reply by LicAC** [reply]
IDM had it since I begun using it! It is really a shame that even the VDH paid version does not have it.

**Comment by joesixpack** [comment]
You can use Google2SRT to download the subtitles or 4KVD to get both the video and subtitles.
